mirrord/config/derive/src/config/field.rs

```diff
@@ -0,0 +1,123 @@
+use proc_macro2_diagnostics::Diagnostic;
+use quote::{quote, ToTokens};
+use syn::{Field, GenericArgument, Ident, PathArguments, Type, Visibility};
+
+use crate::config::flag::{ConfigFlags, ConfigFlagsType, EnvFlag};
+
+#[derive(Debug)]
+pub struct ConfigField {
+    ident: Option<Ident>,
+    option: Option<Type>,
+    ty: Type,
+    vis: Visibility,
+    flags: ConfigFlags,
+}
+
+impl ConfigField {
+    fn is_option(field: &Field) -> Option<Type> {
+        let seg = if let Type::Path(ty) = &field.ty {
+            ty.path.segments.first()?
+        } else {
+            return None;
+        };
+
+        if seg.ident != "Option" {
+            return None;
+        }
+
+        match &seg.arguments {
+            PathArguments::AngleBracketed(generics) => match generics.args.first() {
+                Some(GenericArgument::Type(ty)) => Some(ty.clone()),
+                _ => None,
+            },
+            _ => None,
+        }
```

Comment:
Again, my preference is for expressions, so I would suggest something like:

```suggestion
(seq.ident == "Option").then(|| {
    if let PathArguments::AngleBracketed(Some(GenericArgument::Type(ty))) = &seq.arguments {
        Some(ty.clone())
    } else {
        None
    }
})       
```